Core/src/org/sleuthkit/autopsy/filesearch/DataSourceFilter.java

```diff
@@ -0,0 +1,69 @@
+/*
+ * Autopsy Forensic Browser
+ *
+ * Copyright 2018 Basis Technology Corp.
+ * Contact: carrier <at> sleuthkit <dot> org
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.sleuthkit.autopsy.filesearch;
+
+import java.awt.event.ActionListener;
+import org.openide.util.NbBundle.Messages;
+
+/**
+ * Filter by mime type used in filter areas of file search by attribute.
+ */
+class DataSourceFilter extends AbstractFileSearchFilter<DataSourcePanel> {
+
+    public DataSourceFilter(DataSourcePanel component) {
+        super(component);
+    }
+
+    public DataSourceFilter() {
+        this(new DataSourcePanel());
+    }
+
+    @Override
+    public boolean isEnabled() {
+        return this.getComponent().isSelected();
+    }
+
+    @Override
+    public String getPredicate() throws FilterValidationException {
+        String predicate = "";
+        for (Long dataSourceObjId : this.getComponent().getDataSourcesSelected()) {
+            predicate += "data_source_obj_id = '" + dataSourceObjId + "' OR ";
+        }
+        if (predicate.length() > 3) {
```

Comment:
This seems like a somewhat confusing way to ensure that you don't have a stray " OR " at the end. It might be clearer to do a test of whether predicate is empty before adding to it in the for loop - if predicate is not empty, append an " OR " before adding the next data source id.